src/services/user.ts

```diff
@@ -155,36 +174,33 @@ type MealPlanComponent = {
 
 export type ProfileImages = { def: string; pref?: string };
 
-const isStudent = (profile: UnformattedProfileInfo): profile is UnformattedStudentProfileInfo => {
-  return (profile as UnformattedStudentProfileInfo).OnOffCampus !== undefined;
-};
+const isStudent = (profile: UnformattedProfileInfo): profile is UnformattedStudentProfileInfo =>
+  Boolean((profile as UnformattedStudentProfileInfo)?.Class);
 
-const isStaff = (profile: UnformattedProfileInfo): profile is UnformattedStaffProfileInfo => {
-  return (profile as UnformattedStaffProfileInfo).Dept !== undefined;
-};
+const isFacStaff = (profile: UnformattedProfileInfo): profile is UnformattedFacStaffProfileInfo =>
+  Boolean((profile as UnformattedFacStaffProfileInfo)?.Dept);
+
+const isAlumni = (profile: UnformattedProfileInfo): profile is UnformattedAlumniProfileInfo =>
+  Boolean((profile as UnformattedAlumniProfileInfo)?.ClassYear);
```

Comment:
Not exactly. This is how we determine what formatting should be applied to the profile info. The only place these functions are used is within this service, in order to apply formatting to profile info only if it exists on the current profile. So these properties are used as representative of what kind of info is present on a profile - because we have no other dependable way of determining that.

Reply:
Perhaps we should add an issue for that. There must be a better way to glean that information and bring it up from the database.

Reply:
Honestly, this is about the best way to handle this in TypeScript, as far as I can tell. It works quite well - if a profile has data that only belongs to one kind of profile, then it is that kind of profile.

The only other, technically more correct way of handling this would be to make `Account` be the cartesian product of possible account types, which would result in some real spaghetti types, and would in the end just add some indirectness to this same assumption.

Reply:
I just think we should pass the account type as a value from the api which has a fuller view of the data rather than making a (good) guess at what the account should be. Even if it works well.

Reply:
So we do currently pass account type from the API, and it has been the source of many problems in the past. Most often, the problem is that what data an account does or does not have is much more complex than just whether or not someone is e.g. an Alumni. 

Realistically, the most responsible solution might be to make all the not-common properties optional (since that's effectively how they are in the API anyways), so that we at least enforce that people check that they exist before they use them. I think it may have been a well-intentioned but ultimately mis-guided approach to try to separate Student Accounts from Alumni Accounts from FacStaff accounts.